### ISSUE TITLE
Lack of memory

### ISSUE THREAD

**Issue by rcaceiro** [athombv/apps-sdk-issues#371]
Hello Athom,

I am trying to develop an app to bridge homey devices to Matter, like Homekitty or you Homekit experiment.

Right now I have a very low amount of memory to work with, I am using project-chip server, it isn't my implementation or of someone on the internet.

At start on the server and bridge itself it uses 120MB of memory (RSS) and about 80-90MB (PSS), when I add some device it crashes with heap error.
It is not possible to expand those limits? I have only this dependency + homey's sdk.

I really would like to continue this project.

**Comment by rcaceiro** [comment]
There is a way of configure --max-old-space-size ?

With some testing I be able of managing the consumption rate.

**Comment by robertklep** [comment]
@rcaceiro I'm fairly sure your app gets killed by Homey's SDK "watchdog", using `--max-old-space-size` wouldn't fix this.

You can _try_ to do some preemptive garbage collecting (`globalThis.gc()`), but it sounds like your issue is caused by the server/bridge just taking up more RAM than is allowed for a Homey app.

**Comment by rcaceiro** [comment]
Hi @robertklep actually it helps to force the gc to run ... but yes that lib is new, makers are focus on implement features, they will optmize later.
The memory consuption is because a lot of things is happen but running when it warms up e the garbage collector runs the memory stabalizes and it is possible keep low ... the the startup is what is killing the process.

**Comment by rcaceiro** [comment]
I've tryied anyway and the problem persists the warm up is a little expensive it reaches 120-130 MB, then estabilizes arround 90-100MB

**Comment by robertklep** [comment]
120MB is the hard limit that the SDK imposes. Even if your app settles on 90-100MB after startup it's still awfully close to that limit.

**Comment by rcaceiro** [comment]
Yes I know, but for now I cannot work around it, I already set flags, run GC at the server start and initialization and it is so close that homey kills it.
Dam, for small things is enough, for more complex things is nothing.

I needs also to node runs with a special flag in order GC run, I don't know if the GC runs or if is unable to reclaim memory

**Comment by robertklep** [comment]
@rcaceiro if the special flag wasn't set you wouldn't be able to run `globalThis.gc()`, so it's enabled already. But running garbage collection is only useful if there is actual memory to reclaim, and it sounds like there isn't.

**Comment by rcaceiro** [comment]
It has after the process of booting up. 
In my computer o see it raise above 150mb and then e falls bellow 90, the warm up is too aggressive.
If I force gc to run is faster to reclaim memory, but the setting up is too expensive for now.
